### ISSUE TITLE
Training logs - reproduction yields worse performance

### ISSUE THREAD

**Issue by helen1c** [RobinWu218/SimDINO#13]
Hi @RobinWu218,

I recently retrained Simdinov2 using the ViT-B-16 model, following the exact configuration from the README, exact the same package versions you suggest in requirements, and the same hyperparameters (same number of GPUs, batch size, etc.). My evaluation results were 76.4 for kNN and 78.2 linear, which is 1.6 points lower than yours.  Could you please share your training logs? This will help me compare the runs and identify any differences.

Cheers! :)

**Comment by RobinWu218** [comment]
This is indeed a bit strange. We attached the log [here](https://github.com/RobinWu218/SimDINO/blob/main/assets/SimDNIOv2-vitb16.json) for your reference. Let me know if you notice differences. Thanks a lot for reporting the issue!

**Comment by helen1c** [comment]
Comparison here.. Any ideas what could be wrong?

![Image](https://github.com/user-attachments/assets/80dc5883-e425-475a-96d7-527e2bf3b08b)

Thanks! :)

**Comment by RobinWu218** [comment]
Can you provide your specific settings in config.yaml? From the plot, the main difference is in the global-local similarity loss (`dino_mcr_comp_loss`). Empirically, we also find that a higher similarity value usually translates to better final performance.

**Comment by helen1c** [comment]
Here you go: [config.txt](https://github.com/user-attachments/files/19244070/config.txt). It is .txt as.yaml is not allowed :)

Please share yours as well.

**Comment by RobinWu218** [comment]
We noticed some differences that we missed during cleanup. I have uploaded a fix to the config, and am reproducing on our side to make sure the results are consistent. Feel free to try it. Let me know if there are problems.

**Comment by helen1c** [comment]
Unfortunately, I got the message that optimizes "soap" is not implemented in timm. I cannot even run the code now.  

Hyperparameters (drop path rate, optimizer) should be updated in the paper. 

In the paper you mentioned that you use AdamW. Did you retrain DinoV2 with soap or AdamW?

**Comment by RobinWu218** [comment]
Ah sorry about that, we didn't put soap in this repo as it performed basically on par with other optimizers that we tried (i.e. accuracy would round up to the same decimal), both on original dinov2 and our method. The only difference then is drop path rate, which we missed and will update in a revision. I reverted the optimizer and you should be able to run now.

**Comment by helen1c** [comment]
So, drop path rate from 0.3 -> 0.2 are responsible for cca 1.5 points? That's indeed interesting. I will try to reproduce results now, and reach out once when training is over. 

Are there any additional insights, which hyperparameters influence the final accuracy the most? 
Do you have plans to scale up pre-training datasets?

Thanks! :)

**Comment by RobinWu218** [comment]
In terms of SimDINO, the balancing parameter [gamma](https://github.com/RobinWu218/SimDINO/blob/8149fc4ed8ee3fd97b7bde900447414e6669e71b/simdinov2/loss/dino_clstoken_loss.py#L158) and `--eps` are worth adjusting for different settings (e.g. varying batch sizes). 

Yes, we have plans to scale up, but we are resource-constrained at the moment.

**Comment by helen1c** [comment]
Few last questions. 

Did you try to increase training resolution? What changes in config files are necessary to do that? For example, training on 518x518 crops like Dinov2.


Did you try Simdinov2 with registers? 

Thanks! :)

**Comment by RobinWu218** [comment]
No we didn't try increasing the resolution. You should be able to achieve that by modifying crops.global_crops_size.

Regarding registers, our released experiments all use 4 registers during training.

**Comment by nengwp** [comment]
If there are any new experimental results, I am looking forward to your sharing
